ui/app/AppLayouts/Browser/BrowserLayout.qml

```diff
@@ -22,115 +22,119 @@ import "../Chat/popups"
 
 // Code based on https://code.qt.io/cgit/qt/qtwebengine.git/tree/examples/webengine/quicknanobrowser/BrowserWindow.qml?h=5.15 
 // Licensed under BSD
-
 Rectangle {
     id: browserWindow
 
     property var globalStore
-    property Item currentWebView: tabs.currentIndex < tabs.count ? tabs.getTab(tabs.currentIndex).item : null
 
-    property Component browserDialogComponent: BrowserDialog {
-        onClosing: destroy()
+    function openUrlInNewTab(url) {
+        var tab = _internal.addNewTab()
+        tab.item.url = _internal.determineRealURL(url)
     }
 
-    property Component jsDialogComponent: JSDialogWindow {}
+    QtObject {
+        id: _internal
 
-    property bool currentTabConnected: false
+        property Item currentWebView: tabs.currentIndex < tabs.count ? tabs.getTab(tabs.currentIndex).item : null
 
-    property Component accessDialogComponent: BrowserConnectionModal {
-        currentTab: tabs.getTab(tabs.currentIndex) && tabs.getTab(tabs.currentIndex).item
-        x: appLayout.width - width - Style.current.halfPadding
-        y: browserWindow.y + browserHeader.height + Style.current.halfPadding
-    }
+        property Component browserDialogComponent: BrowserDialog {
+            onClosing: destroy()
+        }
 
-    // TODO we'll need a new dialog at one point because this one is not using the same call, but it's good for now
-    property Component sendTransactionModalComponent: SignTransactionModal {
-        store: browserWindow.globalStore
-    }
+        property Component jsDialogComponent: JSDialogWindow {}
 
-    property Component signMessageModalComponent: SignMessageModal {}
+        property Component accessDialogComponent: BrowserConnectionModal {
+            currentTab: tabs.getTab(tabs.currentIndex) && tabs.getTab(tabs.currentIndex).item
+            // the 78 is for width of the navBar
+            x: browserWindow.width - width - Style.current.halfPadding + 78
```

Comment:
We might want to use a constant or a dynamic value here, in case the navBar changes someday

Reply:
Found a better way to do it :) thanks